Fixed #35623 -- Documented that a field cannot be named 'check'. (#18465)
django/django: docs/topics/db/models.txt

#. A field name cannot end with an underscore, for similar reasons.

> #. Additionally, the word check is a reserved keyword in Django and cannot be
>    used as a field name. This is because check is a special method in Django models.
>    To work around this limitation, consider using another field name, or you can use
>    db_column as described below. For example::
> 
> 
>        class Example(models.Model):
>            field_check = models.IntegerField(db_column="check")
>            # Use 'db_column' to retain 'check' in the database.

Reviewer: To follow the structure of other bullets, I think we should have something like:
```suggestion
#. A field name cannot be 'check', as this would override the check framework's
   Model.check() method.
```

Author: Thank you for your suggestion! I've incorporated your changes into the pull request.

#. A field name cannot end with an underscore, for similar reasons.

#. Additionally, the word check is a reserved keyword in Django and cannot be
   used as a field name. This is because check is a special method in Django models.
   To work around this limitation, consider using another field name, or if you
   need to keep check as the database column name, use the :attr:`~Field.db_column`
   option. For example::

>        class Example(models.Model):

Reviewer 2: This is now a bit redundant with the reference to `db_column` directly below this. In my opinion it's not needed.

Author: Thank you. I removed the additional explanations.
